drivers/serial/uart_hpmicro.c

```diff
@@ -0,0 +1,322 @@
+
+	return 0;
+}
+
+static int uart_hpm_poll_in(const struct device *dev, unsigned char *c)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	if (status_success != uart_receive_byte((UART_Type *)cfg->base, c)) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+static void uart_hpm_poll_out(const struct device *dev, unsigned char c)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	uart_send_byte((UART_Type *)cfg->base, c);
+}
+
+static int uart_hpm_err_check(const struct device *dev)
+{
+	int errors = 0;
+
+	return errors;
+}
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+int uart_hpm_fifo_fill(const struct device *dev, const uint8_t *tx_data,
+			 int len)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+	uint8_t num_tx = 0U;
+
+	if (uart_check_status((UART_Type *)cfg->base, uart_stat_transmitter_empty) != 0) {
+		while ((len - num_tx > 0) && (num_tx < 16)) {
+			uart_send_byte((UART_Type *)cfg->base, tx_data[num_tx++]);
+		}
+	}
+
+	return num_tx;
+}
+
+int uart_hpm_fifo_read(const struct device *dev, uint8_t *rx_data,
+			 const int size)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+	uint8_t num_rx = 0U;
+
+	while ((size - num_rx > 0) &&
+		(uart_check_status((UART_Type *)cfg->base, uart_stat_data_ready) != 0)) {
+		uart_receive_byte((UART_Type *)cfg->base, &rx_data[num_rx++]);
+	}
+
+	return num_rx;
+}
+
+void uart_hpm_irq_tx_enable(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	uart_enable_irq((UART_Type *)cfg->base, uart_intr_tx_slot_avail);
+}
+
+void uart_hpm_irq_tx_disable(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	uart_disable_irq((UART_Type *)cfg->base, uart_intr_tx_slot_avail);
+}
+
+int uart_hpm_irq_tx_ready(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	if (uart_check_status((UART_Type *)cfg->base, uart_stat_transmitter_empty) != 0) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+int uart_hpm_irq_tx_complete(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	if (uart_check_status((UART_Type *)cfg->base, uart_stat_transmitter_empty) != 0) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+void uart_hpm_irq_rx_enable(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	uart_enable_irq((UART_Type *)cfg->base, uart_intr_rx_data_avail_or_timeout);
+}
+
+void uart_hpm_irq_rx_disable(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	uart_disable_irq((UART_Type *)cfg->base, uart_intr_rx_data_avail_or_timeout);
+}
+
+int uart_hpm_irq_rx_ready(const struct device *dev)
+{
+	const struct uart_hpm_cfg *const cfg = dev->config;
+
+	if (uart_check_status((UART_Type *)cfg->base, uart_stat_data_ready) != 0) {
+		return 1;
+	} else {
+		return 0;
+	}
+}
+
+void uart_hpm_irq_err_enable(const struct device *dev)
+{
+
+}
+
+void uart_hpm_irq_err_disable(const struct device *dev)
+{
+
+}
```

Comment:
A small comment would be nice here to explain why these functions are empty. I assume it is because this UART driver doesn't support these features.

Reply:
I don't think this had been addressed?

Reply:
Sorry, soc doesn't provide this function, should I keep this function or add a comment. Currently my approach is to add comments.